### PR TITLE
add arcUp arcDown axis modes

### PR DESCRIPTION
**Pull request by jafu888**
Added arcUp and arcDown modes to arcMode 
This is to support Motion() where the start is where ever you are from.

image

https://user-images.githubusercontent.com/15019413/192195008-9e8aee62-2179-4f98-9d67-499f20920b6c.mov

### CODE REVIEW

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionPaths.java [hunk 1]
```diff
@@ -120,6 +120,62 @@ void initCartesian(KeyPosition c, MotionPaths startTimePoint, MotionPaths endTim
...
+        point.mMode = MotionPaths.CARTESIAN;
+        point.mX = (int) (startTimePoint.mX + pathVectorX * dxdx + pathVectorY * dxdy
+                - scaleX * scaleWidth / 2);
+        point.mY = (int) (startTimePoint.mY + pathVectorX * dydx + pathVectorY * dydy
+                - scaleY * scaleHeight / 2);
+
+        point.mKeyFrameEasing = Easing.getInterpolator(c.mTransitionEasing);
+        point.mPathMotionArc = c.mPathMotionArc;
+    }
     /**
```
Comment (oscar-ad) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionPaths.java:179]: Please leave a line before the next method.

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/utils/ArcCurveFit.java [hunk 2]
```diff
@@ -275,6 +281,15 @@ public ArcCurveFit(int[] arcModes, double[] time, double[][] y) {
                     break;
                 case ARC_START_LINEAR:
                     mode = START_LINEAR;
+                    break;
+                case ARC_UP:
+                    Utils.log("Arc ARC_UP");
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/utils/ArcCurveFit.java:286]: Remove or wrap in something like `if(DEBUG)`

**Review by oscar-ad — CHANGES_REQUESTED** [review]
Don't forget to reformat files.

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/utils/ArcCurveFit.java (hunk 2 above)
Comment (jafu888) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/utils/ArcCurveFit.java]: removed

**Review by oscar-ad — APPROVED** [review]

**Pull request merged**